packages/qdrant-loader/src/qdrant_loader/connectors/factory.py

```
@@ -3,13 +3,27 @@
 
     if not connector_class:
         raise ValueError(f"Unsupported connector type: {key}")
 
-    return connector_class(config)
+    # Try to construct with checkpoint_cursor kwarg; fall back to single-arg.
+    try:
+        # Prefer keyword to be explicit; works for connectors that declare
+        # `checkpoint_cursor` as a parameter (positional or keyword).
+        return connector_class(config, checkpoint_cursor=checkpoint_cursor)
+    except TypeError:
+        # Connector likely doesn't accept the extra arg – fall back.
```

Ruff (0.15.15)

[warning] Comment contains ambiguous `–` (EN DASH). Did you mean `-` (HYPHEN-MINUS)?

(RUF003)